### PR TITLE
Update Error cause

### PR DESCRIPTION
**Pull request by clover2123**
* https://tc39.es/proposal-error-cause
* Remove redundant defineOwnPropertyThrowsExceptionWhenStrictMode method

### CODE REVIEW

File: src/runtime/EnvironmentRecord.h [hunk 1]
```diff
@@ -240,13 +240,13 @@ class ObjectEnvironmentRecord : public EnvironmentRecord {
             ObjectPropertyDescriptor::PresentAttribute attribute = (ObjectPropertyDescriptor::PresentAttribute)(ObjectPropertyDescriptor::WritablePresent | ObjectStructurePropertyDescriptor::EnumerablePresent);
             if (canDelete)
                 attribute = (ObjectPropertyDescriptor::PresentAttribute)(attribute | ObjectPropertyDescriptor::ConfigurablePresent);
-            m_bindingObject->defineOwnPropertyThrowsExceptionWhenStrictMode(state, name, ObjectPropertyDescriptor(Value(), attribute));
```
Comment (ksh8281) [inline src/runtime/EnvironmentRecord.h:243]: Why do you modify these lines?

Comment (ksh8281) [reply src/runtime/EnvironmentRecord.h:243]: IMO some defineOwnPropertyThrowsExceptionWhenStrictMode for needs to support legacy functions

Comment (clover2123) [reply src/runtime/EnvironmentRecord.h:243]: I'm not sure that `defineOwnPropertyThrowsExceptionWhenStrictMode` is really necessary or not.
I can only find the following spec in the latest standard (https://tc39.es/ecma262/#sec-definepropertyorthrow).
> 7.3.8 DefinePropertyOrThrow ( O, P, desc )
1. Assert: Type(O) is Object.
2. Assert: IsPropertyKey(P) is true.
3. Let success be ? O.[[DefineOwnProperty]](P, desc).
4. If success is false, throw a TypeError exception.
5. Return success.

But it seems that there is no specification about `defineOwnPropertyThrowsExceptionWhenStrictMode`.
Would you please let me know about its usage for legacy spec?

Comment (ksh8281) [reply src/runtime/EnvironmentRecord.h:243]: The most horrible things of legacy is what there is no document about they want
How about run programs with node/browser?
If there is no error, we can change it as spec said
If there is a error, we can add a internal test case for it

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**